### ISSUE TITLE
Rewrite MethodHandles.Lookup.checkClassAccess to match spec

### ISSUE THREAD

**Issue by DanHeidinga** [eclipse/openj9#3047]
>A class or interface C is accessible to a class or interface D if and only if one of the following is true:
• C is public, and a member of the same run-time module as D (§5.3.6).
• C is public, and a member of a different run-time module than D, and C's run- time module is read by D's run-time module, and C's run-time module exports C's run-time package to D's run-time module.
• C is not public, and C and D are members of the same run-time package.

http://cr.openjdk.java.net/~iris/se/11/latestSpec/java-se-11-jvms-draft-diffs.pdf

_Originally posted by @DanHeidinga in https://github.com/eclipse/openj9/issues/3037#issuecomment-424706869_

**Comment by DanHeidinga** [comment]
Based on that, our implementation is incorrect. There should be two cases in that code:

When the targetClass is public, do the larger module related check
When targetClass is not public - check they are the same package and the lookup has package access

**Comment by DanHeidinga** [comment]
`checkClassModuleVisibility()` is missing the `MODULE` lookupMode as well.

**Comment by pshipton** [comment]
Not sure if this will make it into the 0.11.0 release. As the release will not wait on this, I'm going to move it to the 0.12.0 milestone. If it does get resolved we can move it back.

**Comment by pdbain-ibm** [comment]
@DanHeidinga I have a sketch of this.  Would you like me to work on it further?

**Comment by DanHeidinga** [comment]
Yes please, especially if it helps resolves some of the JTREG issues we've been focused on over the past week

**Comment by pdbain-ibm** [comment]
Okay, will do.

**Comment by pshipton** [comment]
Doesn't seem to be anybody working on this now, Moving forward to the next release.
